### ISSUE TITLE
Appium not selecting correct device

### ISSUE THREAD

**Issue by Floodzie2** [appium/appium#5488]
Hi,

I am having trouble getting Appium to select the correct device. When launching (I launch from the command line) it keeps trying to add 'Simulator' to the device name. My devices show up under the SIMULATORS section in XCode->Devices, but since upgrading to XCode 7 they do not show with 'Simulator' appended to the name when listing devices from the command line.

$ xcodebuild -version
Xcode 7.0
Build version 7A176x

$ appium -v
1.4.10

$ instruments -s devices
Known Devices:
iPad 2 (9.0) [0A381C47-2444-4145-B30F-B737E0465D8D]
iPhone 6 (9.0) [11723F76-802C-4B57-A99F-4124C637CEBD]
iPhone 6 Plus (9.0) [B723210E-130D-46B2-B416-0518A65633BF]

$ appium  --command-timeout 600 --show-sim-log --app /Users/Shared/Jenkins/workspace/TEST-smoke-test/mobile/soteria-mobile/build/Soteria-Mobile.app --device-name "iPad 2" --platform-name iOS --force-ipad  --full-reset -l
info: [debug] Starting Appium in pre-launch mode
info: Pre-launching app
info: [debug] Using local app from command line: /Users/Shared/Jenkins/workspace/TEST-smoke-test/mobile/soteria-mobile/build/Soteria-Mobile.app
info: [debug] Creating new appium session 0ffb6b94-c4c0-4d13-b9a0-6c1459cceb73
info: [debug] Removing any remaining instruments sockets
info: [debug] Cleaned up instruments socket /tmp/instruments_sock
info: [debug] Setting Xcode version
info: [debug] Xcode version set to 7.0
info: [debug] Setting iOS SDK Version
info: [debug] iOS SDK Version set to 9.0
info: [debug] Checking whether instruments supports our device string
info: [debug] Getting list of devices instruments supports
info: [debug] Instruments is at: /Applications/Xcode.app/Contents/Developer/usr/bin/instruments
info: [debug] Getting device string from opts: {"forceIphone":false,"forceIpad":true,"xcodeVersion":"7.0","iOSSDKVersion":"9.0","deviceName":"iPad 2","platformVersion":null}
info: [debug] fixDevice is on
info: [debug] Final device string is: 'iPad 2 (9.0 Simulator)'
error: Could not find a device to launch. You requested 'iPad 2 (9.0 Simulator)', but the available devices were: ["iPad 2 (9.0) [0A381C47-2444-4145-B30F-B737E0465D8D]","iPhone 6 (9.0) [11723F76-802C-4B57-A99F-4124C637CEBD]","iPhone 6 Plus (9.0) [B723210E-130D-46B2-B416-0518A65633BF]"]
info: [debug] Cleaning up appium session
error: Could not pre-launch appium: Error: Could not find a device to launch. You requested 'iPad 2 (9.0 Simulator)', but the available devices were: ["iPad 2 (9.0) [0A381C47-2444-4145-B30F-B737E0465D8D]","iPhone 6 (9.0) [11723F76-802C-4B57-A99F-4124C637CEBD]","iPhone 6 Plus (9.0) [B723210E-130D-46B2-B416-0518A65633BF]"]

Changing the device name from 'iPad 2' to 'iPad' does not work either.

Under my previous XCode the devices listed like so:

$ xcodebuild -version
Xcode 6.4
Build version 6E35b

$ instruments -s devices
Known Devices:
iPad 2 (8.4 Simulator) [8547882D-6F8A-4DFA-8F6F-CCE4C10DFB2A]

As you can see, 'Simulator' is added to the name in XCode 6.4 but not since upgrading to XCode 7.

Anyone any ideas? Thanks in advance!

**Comment by sebv** [comment]
@moizjv do we support xCode7 already?

**Comment by darraghs** [comment]
Hi, 
I am working with Floodzie2 and looked at the issue, we are using Appium 1.4.10 with Xcode 7 beta 5 .

Here is the diff of the change I had to make to get Appium to select the iPad simulator

--- ./appium/lib/devices/ios/ios.js.bak	2015-08-11 10:41:39.000000000 +0100
+++ appium/lib/devices/ios/ios.js	2015-08-11 10:25:17.000000000 +0100
@@ -1008,7 +1008,8 @@
       'iPad Simulator (8.3 Simulator)': 'iPad 2 (8.3 Simulator)',
       'iPad Simulator (8.4 Simulator)': 'iPad 2 (8.4 Simulator)',
       'iPad Simulator (7.1 Simulator)': 'iPad 2 (7.1 Simulator)',
-      'iPad Simulator (9.0 Simulator)': 'iPad 2 (9.0 Simulator)',
+      //'iPad Simulator (9.0 Simulator)': 'iPad 2 (9.0 Simulator)',
+      'iPad (9.0 Simulator)': 'iPad 2 (9.0) [019546C5-2B29-478E-9968-78EEBE5FF2A9]',
       'iPhone Simulator (8.4 Simulator)': 'iPhone 6 (8.4 Simulator)',
       'iPhone Simulator (8.3 Simulator)': 'iPhone 6 (8.3 Simulator)',
       'iPhone Simulator (8.2 Simulator)': 'iPhone 6 (8.2 Simulator)',
@@ -1017,6 +1018,7 @@
       'iPhone Simulator (9.0 Simulator)': 'iPhone 6 (9.0 Simulator)',
       'iPhone Simulator (7.1 Simulator)': 'iPhone 5s (7.1 Simulator)'
     };
+    logger.debug("HERE: "+iosDeviceString+", found: " + CONFIG_FIX[iosDeviceString]);
     if (CONFIG_FIX[iosDeviceString]) {
       var oldDeviceString = iosDeviceString;
       iosDeviceString = CONFIG_FIX[iosDeviceString];:


Here is the output 
info: [debug] fixDevice is on
info: [debug] HERE: iPad (9.0 Simulator), found: iPad 2 (9.0) [019546C5-2B29-478E-9968-78EEBE5FF2A9]
info: [debug] Fixing device. Changed from: "iPad (9.0 Simulator)" to: "iPad 2 (9.0) [019546C5-2B29-478E-9968-78EEBE5FF2A9]"
info: [debug] Final device string is: 'iPad 2 (9.0) [019546C5-2B29-478E-9968-78EEBE5FF2A9]'

**Comment by mgehlen** [comment]
any chance there will be a fix for this or can I pull the hotfix from @darraghs from somewhere?

**Comment by singkorn** [comment]
I think the conflict comes from the "(9.0 Simulator)" string.
 
What I did was commented out these two lines in ios.js and run appium with the specific iOS 9 simulator device name (--device-name "iPhone 6 (9.0) [XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX]")

 993   if (opts.iOSSDKVersion >= 8) {
 994    // iosDeviceString += " (" + reqVersion + " Simulator)";
 995   } else if (opts.iOSSDKVersion >= 7.1) {
 996    // iosDeviceString += " - Simulator - iOS " + reqVersion;
 997   }

**Comment by amedvedjev** [comment]
+1 same problem

**Comment by ericwengelking** [comment]
@amedvedjev @singkorn I made this change to get Xcode 7 beta 6 to select correct device
https://github.com/appium/appium/pull/5585

Now I also see instruments crash unless I pass in --native-instruments-lib parameter when starting appium
```
node . --native-instruments-lib
```

**Comment by JasonLunn** [comment]
+1

**Comment by ericwengelking** [comment]
@JasonLunn @amedvedjev @singkorn Here is a pull request to fix the device issue with Xcode 7 beta 6
https://github.com/appium/appium/pull/5586/files

I get the same instruments crash but I can resolve it by passing in parameter `--native-instruments-lib`
```
node . --native-instruments-lib
```

**Comment by JasonLunn** [comment]
@ericwengelking - is there any work being done to resolve the crash without having to enable native instruments? I ask because as I understand it, that has a substantial performance penalty?

**Comment by ericwengelking** [comment]
It looks like that is a separate issue @JasonLunn , this fix at least unblocks running against Xcode 7 beta 6 even though there is a huge performance hit with the --native-instruments-lib parameter

**Comment by ericwengelking** [comment]
This also works for Xcode 7.0 GM prerelease and Xcode 7.1 beta 1

**Comment by ericwengelking** [comment]
@JasonLunn when I use Xcode 7.1 beta I don't get the instruments error

**Comment by ericwengelking** [comment]
I think this can be marked as resolved

**Comment by manishmittal79** [comment]
@ericwengelking / or dev for appium: I am continuously getting instruments crashed. Thought the iOS app is installed on real device but instrument is crashing and not able to start the test run. Can you please let me know what I need to do make it working. I have installed 1.4.11 and running xcode7. The iOS is 8.4.1 and iPhone 5S.

![screen shot 2015-09-21 at 4 58 51 pm](https://cloud.githubusercontent.com/assets/2325064/10004884/11b1dd4a-6082-11e5-8e24-93eeb2b54a0b.png)

**Comment by Akulavenkatesh** [comment]
I tried this for code written using Pycharm, haven't tried this on Appium.app:
1. Open Xcode > Window > Devices 
Copy the device simulator name correctly and paste it in your code. Even a slight change in the name will throw the above error

or

2. If the above didn't work, delete all the simulator devices from Xcode > Window > Devices except the one device you need, then do 
'deviceName': 'iPhone Simulator'

I got this error:
iPhone 6s (9.1)', but the available devices were: ["iPhone 6 (9.1) [9BD1BA7F-1E60-4327-8511-A98816F3C951]

Replaced "iPhone 6s" with "iPhone Simulator" in my code and that solved my problem.

**Comment by JESii** [comment]
Regardless of what I specify in my appium.txt, the device started is always "iPhone 6s Plus / iOS 9.2" so instriuments always crashes with "Timed out waiting for device to boot"

**Comment by SrinivasanTarget** [comment]
@imurchie @scottdixon  This is resolved in our current architecture.Please close this.
